### PR TITLE
partition size metric 

### LINKED ISSUE #267: SBS-MEM-AUDIT and SBS-MEM-LOGS : Support partition_size metric
```erlang
handle_metric_req({partition_size, xyzAudit_86400@_, 0, 0} ReplyFun, State) ->
...
{partition_size, TableAlias, FromPartitionIndex, ToPartitionIndex}
% where CurrentpartitionIndex = 0

% last four days of partition
handle_metric_req({partition_size, xyzAudit_86400@_, -4, -1} ReplyFun, State) ->
...
```

### PR DESCRIPTION
**Pull request by shamis**
fixes #267

### CODE REVIEW

File: src/imem_metrics.erl [hunk 1]
```diff
@@ -65,6 +65,12 @@ handle_metric_req(data_nodes, ReplyFun, State) ->
     State;
 handle_metric_req(process_statistics, ReplyFun, State) ->
     process_statistics(ReplyFun, State);
+handle_metric_req({partition_size, TableAlias, FromPartitionIndex,
+                   ToPartitionIndex}, ReplyFun, State) when is_atom(TableAlias),
+        is_integer(FromPartitionIndex), is_integer(ToPartitionIndex) ->
```
Comment (c-bik) [inline src/imem_metrics.erl]: Please reformat (somthing similar to):
```erlang
handle_metric_req(
    {partition_size, TableAlias, FromPartitionIndex, ToPartitionIndex}, ReplyFun, State
) when is_atom(TableAlias), is_integer(FromPartitionIndex), is_integer(ToPartitionIndex) ->
```

File: src/imem_metrics.erl [hunk 2]
```diff
@@ -131,3 +137,19 @@ max_key(heap_size) -> max_heap_size;
...
+            Tables = lists:sort(imem_meta:physical_table_names(TableAlias)),
+            PartitionList = lists:zip(lists:seq(0, length(Tables) - 1), Tables),
+            lists:foldl(
+                fun({Num, Table}, Acc) when (Num >= FPIndex), (Num =< TPIndex) ->
+                    imem_meta:table_size(Table) + Acc;
+                   (_, Acc) -> Acc
+                end, 0, PartitionList);
+        false ->
+            0
+    end.
```
Comment (c-bik) [inline src/imem_metrics.erl]: Newline

File: src/imem_metrics.erl [hunk 3]
```diff
@@ -131,3 +137,19 @@ max_key(heap_size) -> max_heap_size;
...
+        true ->
+            FPIndex = erlang:abs(FromPartitionIndex),
+            TPIndex = erlang:abs(ToPartitionIndex),
+            Tables = lists:sort(imem_meta:physical_table_names(TableAlias)),
+            PartitionList = lists:zip(lists:seq(0, length(Tables) - 1), Tables),
+            lists:foldl(
+                fun({Num, Table}, Acc) when (Num >= FPIndex), (Num =< TPIndex) ->
+                    imem_meta:table_size(Table) + Acc;
+                   (_, Acc) -> Acc
+                end, 0, PartitionList);
```
Comment (c-bik) [inline src/imem_metrics.erl]: Formatting
```erlang
fun
    ({Num, Table}, Acc) when (Num >= FPIndex), (Num =< TPIndex) ->
        imem_meta:table_size(Table) + Acc;
    (_, Acc) -> Acc
end
```

File: src/imem_metrics.erl [hunk 4]
```diff
@@ -131,3 +137,19 @@ max_key(heap_size) -> max_heap_size;
...
+
+partition_size(TableAlias, FromPartitionIndex, ToPartitionIndex) ->
+    case imem_meta:is_time_partitioned_alias(TableAlias) of
+        true ->
+            FPIndex = erlang:abs(FromPartitionIndex),
+            TPIndex = erlang:abs(ToPartitionIndex),
+            Tables = lists:sort(imem_meta:physical_table_names(TableAlias)),
+            PartitionList = lists:zip(lists:seq(0, length(Tables) - 1), Tables),
+            lists:foldl(
+                fun({Num, Table}, Acc) when (Num >= FPIndex), (Num =< TPIndex) ->
```
Comment (c-bik) [inline src/imem_metrics.erl]: Parentheses `(` or `)` conditions not necessary

**Comment by shamis** [comment]
@c-bik, have added the test cases. please verify

File: test/imem_metrics_ct.erl [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+-include_lib("imem_meta.hrl").
+
+%%====================================================================
+%% Test cases.
+%%====================================================================
+
+test_partition_size(_Config) ->
+    ct:pal(info, ?MAX_IMPORTANCE, ?MODULE_STRING ++ ":test_partition_size/1 - Start ===>~n", []),
+    ct:pal(info, ?MAX_IMPORTANCE, ?MODULE_STRING ++ ":schema ~p~n", [imem_meta:schema()]),
+    ct:pal(info, ?MAX_IMPORTANCE, ?MODULE_STRING ++ ":data nodes ~p~n", [imem_meta:data_nodes()]),
```
Comment (c-bik) [inline test/imem_metrics_ct.erl]: `++` isn't needed!
these are non-distribution tests logging schema or data nodes isn't required

File: test/imem_metrics_ct.erl [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+    PartitionTableAlias = test_5@,
+    ?assertEqual(#{size => 0}, imem_metrics:get_metric({partition_size, test, -10, 0})),
```
Comment (c-bik) [inline test/imem_metrics_ct.erl]: Log what is being tested per section, i.e here "non existing tables/partitions"

File: test/imem_metrics_ct.erl [hunk 7]
```diff
@@ -0,0 +1,80 @@
...
+    ?assertEqual(#{size => 0}, imem_metrics:get_metric({partition_size, PartitionTableAlias, -10, 0})),
+    ?assertMatch({ok, _}, imem_meta:create_check_table(PartitionTableAlias, {record_info(fields, ddLog), ?ddLog, #ddLog{}}, [{record_name, ddLog}, {type, ordered_set}], system)),
+    ?assertEqual(0, imem_meta:table_size(PartitionTableAlias)),
+    ?assertEqual(#{size => 0}, imem_metrics:get_metric({partition_size, PartitionTableAlias, -10, 0})),
```
Comment (c-bik) [inline test/imem_metrics_ct.erl]: "empty partitions"

File: test/imem_metrics_ct.erl [hunk 8]
```diff
@@ -0,0 +1,80 @@
...
+    LogRec = #ddLog{logTime = ?TIME_UID, logLevel = info, pid = self()
+        , module = ?MODULE, function = meta_partitions, node = node()
+        , fields = [], message = <<"some log message">>},
+    [PartitionTable1] = imem_meta:physical_table_names(PartitionTableAlias),
+    ?assertEqual(ok, imem_meta:write(PartitionTable1, LogRec)),
+    ?assertEqual(#{size => 1}, imem_metrics:get_metric({partition_size, PartitionTableAlias, 0, 0})),
```
Comment (c-bik) [inline test/imem_metrics_ct.erl]: Add `?assertEqual(1, imem_meta:table_size(PartitionTableAlias)),` before

File: test/imem_metrics_ct.erl [hunk 9]
```diff
@@ -0,0 +1,80 @@
...
+    ct:pal(info, ?MAX_IMPORTANCE, ?MODULE_STRING ++ [io_lib:format(" table : ~p size : ~p ", [T, imem_meta:table_size(T)]) || T <- imem_meta:physical_table_names(PartitionTableAlias)]),
+    imem_meta:drop_table(PartitionTable2),
+    ?assertEqual(#{size => 2}, imem_metrics:get_metric({partition_size, PartitionTableAlias, 0, 0})),
+    ?assertEqual(#{size => 2}, imem_metrics:get_metric({partition_size, PartitionTableAlias, -1, 0})),
+    ?assertEqual(#{size => 6}, imem_metrics:get_metric({partition_size, PartitionTableAlias, -2, 0})),
+    ct:pal(info, ?MAX_IMPORTANCE, ?MODULE_STRING ++ ":dropped ~p~n", [PartitionTable2]),
+    [imem_meta:drop_table(T) || T <- imem_meta:physical_table_names(PartitionTableAlias)],
+    ?assertEqual(true, lists:member({imem_meta:schema(), node()}, imem_meta:data_nodes())),
+    ct:pal(info, ?MAX_IMPORTANCE, ?MODULE_STRING ++ ":~p:test_partition_size~n", [?MODULE]),
+    ct:pal(info, ?MAX_IMPORTANCE, ?MODULE_STRING ++ ":success ~p~n", [partition_size]),
```
Comment (c-bik) [inline test/imem_metrics_ct.erl]: Add comments for each test scenarios describing how test data are generated what is expected and how is it verified

File: src/imem_metrics.erl [hunk 10]
```diff
@@ -131,3 +138,26 @@ max_key(heap_size) -> max_heap_size;
...
+    case imem_meta:is_time_partitioned_alias(TableAlias) of
+        true ->
+            Now = imem_datatype:seconds_since_epoch(imem_meta:time()),
+            [_, _, _, _, PTStr, _, _] = imem_meta:parse_table_name(TableAlias),
+            PTime = list_to_integer(PTStr),
+            TableNames = [imem_meta:partitioned_table_name(TableAlias, {Now + (PTime * T), 0})
+                            || T <- lists:seq(FromPartitionIndex, ToPartitionIndex)],
+            lists:foldl(
+                fun
+                    (Table, Acc) ->
```
Comment (c-bik) [inline src/imem_metrics.erl]: For simple match function multi-line isn't requied:
```erlang
fun(...) ->
    ...
end
```
but
```erlang
fun
    (match_pattern_1) ->
        ...
    (match_pattern_2) ->
        ...
    ...
end
```

File: src/imem_metrics.erl [hunk 11]
```diff
@@ -131,3 +138,26 @@ max_key(heap_size) -> max_heap_size;
...
+                        try imem_meta:table_size(Table) of
+                            Size when is_integer(Size) ->
+                                Size + Acc
```
Comment (c-bik) [inline src/imem_metrics.erl]: https://github.com/K2InformaticsGmbH/imem/blob/d23a87319b97d18f167da51add7b782677f98e2f/src/imem_meta.erl#L2355
`when is_integer...` isn't required!

File: src/imem_metrics.erl [hunk 12]
```diff
@@ -131,3 +138,26 @@ max_key(heap_size) -> max_heap_size;
...
+                        catch
+                            _:_ ->
+                                Acc
+                        end
+                end, 0, TableNames);
```
Comment (c-bik) [inline src/imem_metrics.erl]: `TableNames` has no other use, merge `imem_meta:partitioned_table_name(TableAlias, {Now + (PTime * T), 0})` in `lists:foldl`

**Comment by shamis** [comment]
have made the changes please verify

**Pull request merged**